### PR TITLE
[DDW-360] missing v1 api endpoints integration 6-9

### PR DESCRIPTION
**Pull request by MarcusHurney**
This PR implements V1 endpoints for the follow Ada API requests:

6.) nextAdaUpdate.js

- Contains a check in `request.js` for a 404 response which indicates an update is not available instead of an error.

7.) applyAdaUpdate.js
8.) postponeAdaUpdate.js
9.) adaTestReset.js

### Basics

- [x] PR is updated to the most recent version of target branch (and there are no conflicts)
- [x] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn run flow:test`)
- [x] There are no *lint* errors or warnings (`yarn run lint`)
- [x] Text changes are proofread and approved (Jane Wild)
- [x] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn run storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly documented and commented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-rendering
- [x] Any code that only works in Electron is neatly separated from components

### Testing
- [x] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [x] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/renderer/app/api/ada/lib/v1/request.js [hunk 1]
```diff
@@ -79,13 +79,27 @@ function typedRequest<Response>(
...
           const { statusCode, statusMessage } = response;
+
           if (!body && statusCode >= 200 && statusCode <= 206) {
             // adds status and data properties so JSON.parse doesn't throw an error
             body = `{
               "status": "success",
-              "data": "statusCode: ${statusCode}, statusMessage: ${statusMessage}"
+              "data": "statusCode: ${statusCode} -- statusMessage: ${statusMessage}"
             }`;
+          } else if (statusCode === 400) {
```
Comment (nikolaglumac) [inline source/renderer/app/api/ada/lib/v1/request.js]: @MarcusHurney are you sure the expected statusCode is `400`?
In the docs it says it should be `404` - AKA not-found?


Also, I am a bit worried that any other status code 400/404 will not be properly handled with this if clause. E.g. other endpoints might have status 400 as real error?

**Comment by nikolaglumac** [comment]
@MarcusHurney the reason this endpoint was returning error code 400 is the fact you are running the backend against the old data layer. I have tested it against the new data layer and I got the expected 404 error code. Since we will switch to the new data layer I have updated the code: https://github.com/input-output-hk/daedalus/pull/1079/commits/ffe35381baaa14e003cb36ce7dfb7dfe5a158cc0
Will explain to you how to run the backend in the same way as I do ;)

**Comment by nikolaglumac** [comment]
@MarcusHurney V1 Api endpoint `/api/internal/reset-wallet-state` is still not implemented on the backend side. This means we are unable to run tests with the new data layer.

**Comment by MarcusHurney** [comment]
@nikolaglumac 
Thanks for looking into this! I like this check here:
`options.path === '/api/internal/next-update`.
Much better than mainly checking for strings as truthy values. I know we love Javascript but we can't  leave it alone with our sister.

**Pull request merged**